launcher/modplatform/modrinth/ModrinthInstanceCreationTask.cpp

```diff
@@ -348,11 +381,11 @@ bool ModrinthCreationTask::parseManifest(const QString& index_path,
                 QJsonObject hashes = Json::requireObject(modInfo, "hashes");
                 QString hash;
                 QCryptographicHash::Algorithm hashAlgorithm;
-                hash = Json::ensureString(hashes, "sha1");
-                hashAlgorithm = QCryptographicHash::Sha1;
+                hash = Json::ensureString(hashes, "sha512");
+                hashAlgorithm = QCryptographicHash::Sha512;
                 if (hash.isEmpty()) {
-                    hash = Json::ensureString(hashes, "sha512");
-                    hashAlgorithm = QCryptographicHash::Sha512;
+                    hash = Json::ensureString(hashes, "sha1");
+                    hashAlgorithm = QCryptographicHash::Sha1;
```

Comment:
What's the justification to reverse these?

Reply:
I need sha512 for ensuring modrinth metadata and I did not want to calculate the sha512 again or to store that in a separate variable. The ensure metadata task expects sha512 for modrinth.

Reply:
no issue in particular. just more expensive to use the 512 hash is it wasn't needed (thous it's not *that* much more expensive). I think `sha1` was initially used because curseforge was using sha1  and the code was mostly copied.